### PR TITLE
TLPORTAL-451: global fix for html escape issue

### PR DESCRIPTION
**Pull request by dlhaines**
Ensure that html_escaping will not turn a nil value into an empty string.

### CODE REVIEW

File: server/courselist.rb [hunk 1]
```diff
@@ -458,8 +458,9 @@ def allow_uniqname_override user
     def uniqnameOverride
 
       config_hash = settings.latte_config
+      
       # See if there is a candidate to use as authenticated userid name.
-      uniqname = html_escape(params['UNIQNAME'])
+      uniqname = CourseList.safe_html_escape(params['UNIQNAME'])
```
Comment (zqian) [inline server/courselist.rb]: The code is inside "courselist.rb" itself. Do we still need to prefix it with CourseList?

File: server/spec/test_courselist.rb [hunk 2]
```diff
@@ -195,6 +195,26 @@ def test_assemble_course_parameters
     assert_equal correct, encoded, "multiple courses as parameters"
   end
 
+  # check if safe_html_escape respects nil input
+  def test_safe_html_escape
```
Comment (zqian) [inline server/spec/test_courselist.rb]: Shall we test the the string that actually need escaping?

File: server/courselist.rb (hunk 1 above)
Comment (dlhaines) [inline server/courselist.rb]: As it turns out yes, at least when using Sinatra.

File: server/spec/test_courselist.rb (hunk 2 above)
Comment (dlhaines) [inline server/spec/test_courselist.rb]: Zhen, I'm not sure what this comment means.

**Pull request merged**